packages/smooth_app/lib/pages/product/add_new_product_page.dart

```diff
@@ -51,64 +52,81 @@ class _AddNewProductPageState extends State<AddNewProductPage> {
   Widget build(BuildContext context) {
     final AppLocalizations appLocalizations = AppLocalizations.of(context);
     final ThemeData themeData = Theme.of(context);
-    return SmoothScaffold(
-      appBar: AppBar(
-          title: Text(appLocalizations.new_product),
-          automaticallyImplyLeading: !_isProductLoaded),
-      body: Padding(
-        padding: const EdgeInsetsDirectional.only(
-          top: VERY_LARGE_SPACE,
-          start: VERY_LARGE_SPACE,
-          end: VERY_LARGE_SPACE,
-        ),
-        child: Stack(
-          children: <Widget>[
-            SingleChildScrollView(
-              child: Column(
-                crossAxisAlignment: CrossAxisAlignment.start,
-                children: <Widget>[
-                  Text(
-                    appLocalizations.add_product_take_photos_descriptive,
-                    style: themeData.textTheme.bodyText1!
-                        .apply(color: themeData.colorScheme.onBackground),
-                  ),
-                  ..._buildImageCaptureRows(context),
-                  _buildNutritionInputButton(),
-                  _buildaddInputDetailsButton()
-                ],
-              ),
+    final UpToDateProductProvider provider =
+        Provider.of<UpToDateProductProvider>(context);
+    Product product = Product(barcode: widget.barcode);
+    return Consumer<UpToDateProductProvider>(
+      builder: (
+        BuildContext context,
+        Object? value,
+        Widget? child,
+      ) {
```

Comment:
Now we listen twice, that doesn't look right to me

Reply:
I didn't get that, like where exactly listen is being used twice, I guess I am not able to get what u just said 😿

Reply:
Don't put too much energy on `UpToDateProductProvider`, that will virtually be gone by a day of two.

Reply:
> Don't put too much energy on UpToDateProductProvider, that will virtually be gone by a day of two.

Wishing a [bye](https://www.youtube.com/watch?v=5K3E5tLoado&t=27s) to `uptodateproductprovider`, I guess I will have to restudy the state management 😵‍💫

Reply:
@AshAman999 You'll be surprised how simple it becomes...